File: server/models/events.js
const mongoose = require("mongoose");

const eventSchema = mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true,
  },
  backgroundImage: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
    trim: true,
  },
  location: {
    type: String,
    required: true,
    trim: true,
  },
  duration: {
    type: String,
    required: true,
    trim: true,
  },
  punchLine1: {
    type: String,
    required: true,
    trim: true,
  },
  punchLine2: {
    type: String,
    required: true,
    trim: true,
  },
  galleryImage: [
    {
      type: String,
      required: true,
    },
  ],
  category: {
    type: String,
    required: true,
    trim: true,
  },
});

const Event = mongoose.model("Event", eventSchema);
module.exports = Event;

File: lib/models/events.dart
// To parse this JSON data, do
//
//     final event = eventFromJson(jsonString);

import 'dart:convert';

// Event eventFromJson(String str) => Event.fromJson(json.decode(str));

// String eventToJson(Event data) => json.encode(data.toJson());

class Event {
    String title;
    String backgroundImage;
    String description;
    String location;
    String duration;
    String punchLine1;
    String punchLine2;
    List<String> galleryImage;
    String category;

    Event({
        required this.title,
        required this.backgroundImage,
        required this.description,
        required this.location,
        required this.duration,
        required this.punchLine1,
        required this.punchLine2,
        required this.galleryImage,
        required this.category,
    });

    Map<String, dynamic> toMap() {
    return {
      'title': title,
      'backgroundImage': backgroundImage,
      'description': description,
      'location': location,
      'duration': duration,
      'punchLine1': punchLine1,
      'punchLine2': punchLine2,
      'galleryImage': galleryImage,
      'category': category,
    };
  }

    factory Event.fromMap(Map<String, dynamic> map) {
    return Event(
      title: map['title'] ?? '',
      backgroundImage: map['backgroundImage'] ?? '',
      description: map['description'],
      location: map['location'],
      duration: map['duration'] ?? '',
      punchLine1: map['punchLine1'],
      punchLine2: map['punchLine2'],
      galleryImage:  List<String>.from(map['galleryImage']),
      category: map['category'],
    );
  }

  String toJson() => json.encode(toMap());
  

  factory Event.fromJson(String source) =>
      Event.fromMap(json.decode(source));
}
